lib/log.js

```diff
@@ -178,25 +178,59 @@ Lug.prototype.activityEvent = function (event, request, data) {
     return P.resolve()
   }
 
+  // log a flowEvent for all activityEvents
+  this.flowEvent(event, request)
+
+  var info = {
+    event: event
+  }
+
+  if (request.headers['user-agent']) {
+    info.userAgent = request.headers['user-agent']
+  }
+
+  optionallySetService(info, request)
+
+  Object.keys(data).forEach(function (key) {
+    info[key] = data[key]
+  })
+
+  this.logger.info('activityEvent', info)
+  this.statsd.write(info)
+
+  return P.resolve()
```

Comment:
I see your point but also, any callers that were waiting for this promise to be resolved before (I haven't checked to see what they were) were waiting for the async stuff in this method to finish. That isn't happening any more.

Maybe that's fine and certainly the tests aren't reporting any problems, but also it changes the meaning of existent calls to `log.activityEvent`. Before this change they could guarantee that `metricsContext.gather` had completed. After this change they can't.

That might not be a big deal but I thought it was worth raising.

To flip the question round: what is the justification for behaving differently with this PR?